### PR TITLE
Enforce authorization checks

### PR DESCRIPTION
**Pull request by iPegii**

### CODE REVIEW

File: services/headless-lms/server/src/controllers/main_frontend/course_instances.rs [hunk 1]
```diff
@@ -101,7 +101,6 @@ pub async fn point_export(
     .await?;
     let course_instance_id = *course_instance_id;
     let (sender, receiver) = tokio::sync::mpsc::unbounded_channel::<ControllerResult<Bytes>>();
```
Comment (nygrenh) [inline services/headless-lms/server/src/controllers/main_frontend/course_instances.rs:103]:
```suggestion
    let (sender, receiver) = tokio::sync::mpsc::unbounded_channel::<BypassAuthorizationCheckResult<Bytes>>();
```

File: services/headless-lms/server/src/controllers/main_frontend/users.rs [hunk 2]
```diff
@@ -13,7 +13,9 @@ pub async fn get_user(
 ) -> ControllerResult<web::Json<User>> {
     let mut conn = pool.acquire().await?;
     let user = models::users::get_by_id(&mut conn, *user_id).await?;
-    Ok(web::Json(user))
+
+    let token = authorize(&mut conn, Act::Teach, Some(user_id), Res::AnyCourse).await?;
+    token.0.ok(web::Json(user))
```
Comment (nygrenh) [inline services/headless-lms/server/src/controllers/main_frontend/users.rs:18]: token.to_authorized_result(web::Json(user))

**Review by nygrenh — CHANGES_REQUESTED** [review]

**Review by nygrenh — CHANGES_REQUESTED** [review]
Document the new functions/structs.
Write an example to the authorize, and skip_authorize documentation

File: services/headless-lms/server/src/domain/authorization.rs [hunk 3]
```diff
@@ -154,11 +157,30 @@ impl<T: Responder> Responder for AuthorizedResponse<T> {
...
 }
 
-/// Can user_id action the resource?
+/// Verifies that user has permission to Act on given Resource and returns an AuthorizationToken
+///
+/// let token = authorize(&mut conn, Act::Edit, Some(user.id), Res::Page(*page_id)).await?;
+///
+/// We check that the user has permission to Edit the given Page, if succesful we return authorized_ok
+///
+/// token.authorized_ok(web::Json(cms_page_info))
```
Comment (nygrenh) [inline services/headless-lms/server/src/domain/authorization.rs:183]: The authorization token is the only way to return a controller result, and should only be used in controller functions that return a response to the user.

File: services/headless-lms/server/src/domain/csv_export.rs [hunk 4]
```diff
@@ -251,6 +251,7 @@ impl Write for CSVExportAdapter {
     }
 }
 
+/// Takes UnboundedReceiverStream-stream with AuthorizedResponse and returns UnboundedReceiverStream-stream with only Bytes
```
Comment (nygrenh) [inline services/headless-lms/server/src/domain/csv_export.rs:254]: Without this one, actix cannot stream our authorized streams as responses

also maybe an example

**Review by nygrenh — CHANGES_REQUESTED** [review]

**Review by jolampi — APPROVED** [review]

**Review by nygrenh — APPROVED** [review]

**Pull request merged**